test/unit/disposable_common.dart

```diff
@@ -272,12 +271,92 @@ void testCommonDisposable(Func<StubDisposable> disposableFactory) {
     });
 
     test(
-        'should return ManagedStreamSubscription that returns null when an'
+        'should throw if stream completes with an error and there is no '
+        'onError handler', () {
+      runZoned(() {
+        // ignore: close_sinks
+        var controller = new StreamController<Null>();
+        // ignore: cancel_subscriptions
+        var subscription = disposable.listenToStream(controller.stream, (_) {});
+        controller.addError(new Exception('intentional'));
+      }, onError: expectAsync2((error, _) {
+        expect(error, new isInstanceOf<Exception>());
+        expect(error.toString(), 'Exception: intentional');
+      }));
+    });
+
+    test(
+        'should call error handler if stream receives an error '
+        'and there was an onError handler set by listenToStream', () {
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {},
+          onError: expectAsync2((error, _) {
+        expect(error, new isInstanceOf<Exception>());
+        expect(error.toString(), 'Exception: intentional');
+      }));
+      controller.addError(new Exception('intentional'));
+    });
+
+    test(
+        'should call error handler if stream receives an error '
+        'and there was an onError handler set on the subscription', () {
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {});
+      subscription.onError(expectAsync2((error, _) {
+        expect(error, new isInstanceOf<Exception>());
+        expect(error.toString(), 'Exception: intentional');
+      }));
+      controller.addError(new Exception('intentional'));
+    });
+
+    test(
+        'should call onDone callback when controller is closed '
+        'and there was an onDone callback set by listenToStream', () {
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {},
+          onDone: expectAsync0(() {}));
+      controller.close();
+    });
+
+    test(
+        'should call onDone callback when controller is closed '
+        'and there was an onDone callback set on the subscription', () {
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {});
+      subscription.onDone(expectAsync0(() {}));
+      controller.close();
+    });
+
+    test('should unmanage subscription when controller is closed', () async {
```

Comment:
should add a test for `should unmanage subscription when controller closes due to an error`